pkg/storage/s3.go

```diff
@@ -1,38 +1,50 @@
 package storage
 
 import (
+	"crypto/x509"
+	"encoding/pem"
 	"io"
+	url2 "net/url"
 	"path"
 	"time"
 
 	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cloudfront/sign"
 	"github.com/aws/aws-sdk-go/service/s3"
 	"github.com/pkg/errors"
 	"github.com/spf13/afero"
 )
 
 // S3 implements the file storage API using S3.
 type S3 struct {
-	bucket       string
-	keyNamespace string
-	logger       Logger
-	client       *s3.S3
-	fs           *afero.Afero
-	tempFs       *afero.Afero
+	bucket                string
+	keyNamespace          string
+	logger                Logger
+	client                *s3.S3
+	fs                    *afero.Afero
+	tempFs                *afero.Afero
+	assetsDomainName      string
+	cfPrivateKey          string
+	cfPrivateKeyID        string
```

Comment:
```suggestion
        cfPrivateKey          *string
	cfPrivateKeyID        *string
```
I'm thinking these need to be nullable. It's possible we may not include them in the environment if we're only doing memory or filesystem storage. So we shouldn't fail if they are not available. This may also mean changing your logic in `pkg/cli/` to say "If storage backend is cdn these values are required, otherwise ignore".